css/properties/margin-left.json

```diff
@@ -4,6 +4,7 @@
       "margin-left": {
         "__compat": {
           "mdn_url": "https://developer.mozilla.org/docs/Web/CSS/margin-left",
+          "spec_url": "https://drafts.csswg.org/css-flexbox-1/#item-margins",
```

Comment:
I don't think including this is actually appropriate, https://drafts.csswg.org/css-flexbox-1/#item-margins doesn't form part of the definition of any of the margin properties directly. Or should any normative requirement involving a feature be listed for that feature? (I'm skeptical.)

Reply:
>  Or should any normative requirement involving a feature be listed for that feature? (I'm skeptical.)

I don’t think the spec URL https://drafts.csswg.org/css-flexbox-1/#item-margins should be associated with the `margin-left` feature. In general, I think we have many cases where the Specifications tables in MDN have way too may specifications listed. Take a look at https://developer.mozilla.org/en-US/docs/Web/CSS/margin-left#Specifications. That’s especially a problem for CSS features.

I’ve edited many MDN articles to trim references from the Specifications tables that don’t really belong where they are. But there’s not really a systematic way to programmatically identify such cases.

Reply:
Yeah, trying to represent the history of where it was specified and when it changed doesn't seem useful.

To shake out problems with taking an extreme approach, would there be any I assume with allowing exactly one spec URL? What's the canonical good case for listing more than one?

As an aside, I also observe there's a comment per spec there in MDN. Are we betting that we won't need that in BCD?

Reply:
> As an aside, I also observe there's a comment per spec there in MDN. Are we betting that we won't need that in BCD?

Yes. And even in MDN, for the JavaScript docs at least, we stopped including the comment column in the Specifications tables altogether — for any new stuff being added there.

Reply:
> To shake out problems with taking an extreme approach, would there be any I assume with allowing exactly one spec URL? What's the canonical good case for listing more than one?

1. Methods with multiple signatures, and
2. Features with normative requirements in multiple specs

For <span>#</span>1, see specifically https://github.com/mdn/browser-compat-data/pull/2983#issuecomment-431799873 and see https://developer.mozilla.org/docs/Web/API/AudioNode/connect.

The gist of that is, when we have a method with multiple signatures, it’s probably gonna have multiple anchors in the spec, and we’re probably gonna want to include both of those in our data.

It’s arguable that in such cases, specs should ideally have one higher-level section/anchor for the method overall, with subsections/anchors for each signature. But our specs don’t always (or maybe even often) do it that way — and of course it’s not our job to re-design how the specs are structured.

It’s also arguable that if a method had multiple signatures, each signature should have a separate a separate (sub)feature entry in BCD. But we’re not currently doing it that way, and it’s not clear that we really should be.

For <span>#<span>2, see https://github.com/mdn/browser-compat-data/pull/2983#issuecomment-435646110 and see https://tc39.es/ecma262/#sec-array.prototype.tolocalestring and https://tc39.es/ecma402/#sup-array.prototype.tolocalestring

The gist of that: we have details in one spec (ECMA-262) that are necessary for anyone wanting to know the required behavior for any implementation that doesn’t support the requirements in defined in another spec (ECMA-402).

And we have about 7 instances of the same specific case where that’s necessary. See https://github.com/mdn/browser-compat-data/pull/2983#issuecomment-435646110

---

So even though I personally would very much prefer that we could just restrict our `spec_url` property to having only a single URL — the investigation I did seems to indicate we have some non-insignificant cases where we can’t to that.

Reply:
Thanks @sideshowbarker, https://heycam.github.io/webidl/#idl-overloading is a solid case of sometimes needing multiple URLs I agree. Due to the rules of Web IDL they should always be within the same spec though.

I still get the feeling that overall we'll want to reduce the number of spec links, and we'll need some guidelines for "Features with normative requirements in multiple specs" especially.

Reply:
```suggestion
          "spec_url": "https://drafts.csswg.org/css-flexbox/#item-margins",
```